packages/pds/src/pipethrough.ts

```diff
@@ -367,80 +418,135 @@ function handleUpstreamRequestError(
 // Request parsing/forwarding
 // -------------------
 
-type Accept = [name: string, flags: Record<string, string>]
+type Accept = [name: string, flags: { q: number }]
+
+// accept-encoding defaults to "identity with lowest priority"
+const ACCEPT_ENC_DEFAULT = ['identity', { q: 0.001 }] as const satisfies Accept
+const ACCEPT_FORBID_STAR = ['*', { q: 0 }] as const satisfies Accept
 
-function negotiateAccept(
+function negotiateContentEncoding(
   acceptHeader: undefined | string | string[],
-  supported: readonly Accept[],
-): readonly Accept[] {
-  // Optimization: if no accept-encoding header is present, skip negotiation
-  if (!acceptHeader?.length) {
-    return supported
+  preferences: readonly [Accept, ...Accept[]],
+): string {
+  if (!acceptHeader) {
+    // If no accept-encoding header is provided, we will use our own preferences
+    // and decode the response if needed.
```

Comment:
Would some code paths be simplified if treated a missing accept-encoding by just applying a default of `identity`?  It seems as though we go down different code paths depending on whether `accept-encoding` is missing vs. when it's `identity`, but we end-up trying to generate the same result for the consumer (output is not encoded).

Reply:
I wanted to allow the PDS to prefer compressed upstream bodies, even if the client requested for identity. Then I realized that we wouldn't want to decode+re-encode so I only supported that when the client did not specify an encoding as this was much simpler to support. You are right though, this is inconsistent and makes the code much harder to read.

I reverted to doing what you suggested.